nvidia-nano/GiveMeWES.yaml

```diff
@@ -0,0 +1,597 @@
+      args:
+        executable: /bin/bash
+
+    - name: Make the overlayfs 32GB
+      community.general.parted:
+        device: /dev/sda
+        number: 2
+        state: present
+        label: gpt
+        name: system-data
+        part_start: 33556480s #s = sectors
+        part_end: 100665343s 
+        fs_type: ext4
+
+    - name: Make the plugin-data (the rest of the space)
+      community.general.parted:
+        device: /dev/sda
+        number: 3
+        state: present
+        label: gpt
+        name: plugin-data
+        part_start: 100665344s  
+        part_end: 100%
+        fs_type: ext4
+
+    - name: Create a ext4 filesystem on /dev/sda2
+      community.general.filesystem:
+        fstype: ext4
+        dev: /dev/sda2
+
+    - name: Create a ext4 filesystem on /dev/sda3
+      community.general.filesystem:
+        fstype: ext4
+        dev: /dev/sda3
+
+    - name: Unmount /dev/sda1 #Unmount this partition so that mkswap can run
+      ansible.posix.mount:
+        path: /dev/sda1
+        state: unmounted
+
+    - name: Turn on the swap
+      shell: mkswap /dev/sda1 -L ext-swap
+      args:
+        executable: /bin/bash
+
+    - name: Put the swap in the startup partition file #table structure: /dev/sda1 swap swap defaults,nofail 0 0
+      ansible.posix.mount:
+        path: swap
+        src: /dev/sda1
+        fstype: swap
+        opts: defaults,nofail
+        dump: 0
+        passno: 0
+        state: present
+        boot: yes
+
+    - name: Put plugin-data in the startup partition file #table structure: /dev/sda3 /media/plugin-data ext4 defaults,nofail,x-systemd.after=local-fs-pre.target,x-systemd.before=local-fs.target 0 2
+      ansible.posix.mount:
+        path: /media/plugin-data
+        src: /dev/sda3
+        fstype: ext4
+        opts: defaults,nofail,x-systemd.after=local-fs-pre.target,x-systemd.before=local-fs.target
+        dump: 0
+        passno: 2
+        state: present
+        boot: yes
+
+    - name: Unconditionally reboot the machine with all defaults
+      ansible.builtin.reboot: # reboot to mount sda3 (tried mounting without rebooting and kept running into errors)
+
+- name: Install Tools
+  hosts: all
+  tasks:
+    - name: install helpful tools
+      ansible.builtin.apt:
+        pkg:
+        - dnsutils
+        - htop
+        - iftop
+        - iotop
+        - jq
+        - nmap
+        - curl
+        - python3-click
+        - dnsmasq
+
+    - name: Install K3s via curl
+      shell: curl -sfL https://get.k3s.io | INSTALL_K3S_VERSION=v1.20.2+k3s1 INSTALL_K3S_SKIP_ENABLE=true K3S_CLUSTER_SECRET=4tX0DUZ0uQknRtVUAKjt sh -
+      args:
+        executable: /bin/bash
+
+    - name: Install the waggle-nodeid tool
+      ansible.builtin.apt:
+        deb: https://github.com/waggle-sensor/waggle-nodeid/releases/download/v1.0.7/waggle-nodeid_1.0.7_all.deb
+
+    - name: Install waggle-node-hostname #the tool is used for a work around to change the hostname 
```

Comment:
change this to be a loop of `with_items` ansible option to do them all at once?  maybe

Reply:
this will be done in a later update